### PR TITLE
Refactor user flow eKYC solutions

### PR DESCRIPTION
**Pull request by rizkybaihaqy**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

eKYC hit api every single steps

## What is the new behavior?

eKYC now takes every require images and then hit the api at result page. Some changes i made are:
- Add Cam component to take ktp image, if used in device with environment facing camera then it will be used as preferred method. If not, then it will just use the only available camera.
- Now result will be requested at the result page at fourth step.
- Add user activities tracking which hit api every next button press and after submit feedback.
- Add spinner loading at result page in case of late api response

All captured photo are stored at localStorage in the client browser

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: frontend/app/components/templates/EkycPage/EkycPage.tsx [hunk 1]
```diff
@@ -85,17 +121,85 @@ export const EkycPage = ({ serviceId, name, shortDesc, longDesc }: Props) => {
...
+              {!loading && result ? (
+                <AnalyticsResult result={result.ocr_ktp} slug={'ocr-ktp'} />
+              ) : (
+                <Spinner />
+              )}
+            </div>
+
+            <Button
+              color={Color.Primary}
+              onClick={() => nextStep(5)}
```
Comment (gregoriusjimmy) [inline frontend/app/components/templates/EkycPage/EkycPage.tsx:184]: @rizkybaihaqy untuk nextstep disini perlu clear state resultnya. gw udah coba selesain sampai finish terus try again lagi pas sampai step result itu masih nampilin result dari percobaan pertama

File: frontend/app/components/templates/EkycPage/EkycPage.tsx [hunk 2]
```diff
@@ -85,17 +121,85 @@ export const EkycPage = ({ serviceId, name, shortDesc, longDesc }: Props) => {
...
+                <>
+                  <span>{`${Math.trunc(
+                    result.face_match.similarity * 100
+                  )}%`}</span>
+                  <p>{result.face_match.match ? 'Verified' : 'Not Verified'}</p>
+                </>
+              ) : (
+                <Spinner />
+              )}
+            </div>
```
Comment (williammfu) [inline frontend/app/components/templates/EkycPage/EkycPage.tsx:171]: Butuh handling kalau engga ada result (response status ok, tapi result kopong).
Tadi pas coba foto selfie liveness ok, foto ktp nya dikosongin eternal loading mutarrr". Kayanya bisa jadi TO DO buat phase selanjutnya (buat API handlernya juga)

Response body
image

Comment (williammfu) [reply frontend/app/components/templates/EkycPage/EkycPage.tsx]: resolved on commit be48a1d

File: frontend/app/components/templates/EkycPage/EkycPage.tsx (hunk 1 above)
Comment (williammfu) [reply frontend/app/components/templates/EkycPage/EkycPage.tsx]: resolved on commit 93e72de

**Review by williammfu — APPROVED** [review]

**Review by gregoriusjimmy — APPROVED** [review]

**Pull request merged**